modules/data/webadmin/tmpl/add_edit_network.tmpl

```diff
@@ -213,6 +213,15 @@
 			<? ENDLOOP ?>
 					</tbody>
 				</table>
+				<script>
+				$("#channels").sortable({
```

Comment:
That's correct. I will implement a non-JS interface.

Reply:
I'd like to rework the channels interface by copying the design of the network servers list.

Instead of the current table, the non-JS UI would be a textarea where each line specifies the configuration of a channel. With JS, the textarea would be replaced by a table with more user friendly inputs and draggable rows. I would also remove the add/edit channel page and move its options to the new interface.

Alternatively I could add a textarea (hidden with JS) to specify the channels in the desired order. This approach would not change the existing UI.

I believe the first approach is cleaner. What are your thoughts?

Reply:
> I'd like to rework the channels interface by copying the design of the network servers list.

Tbh, I'm not entirely happy with what I did there. Currently the JS editor edits the (hidden) text area, while it would probably be cleaner to send a flag to the server whether to take input from the text area or from separate fields resulting from JS editor directly. Not sure.

> the non-JS UI would be a textarea where each line specifies the configuration of a channel

The channel config is more complicated than list of servers, so this would make webadmin essentially unusable for editing channels without JS. Maybe you have a mock of the new UI if I misunderstood what do you mean?

> I would also remove the add/edit channel page and move its options to the new interface.

This will break modules which add parts of UI to that page, e.g. stickychan. How would you present that in a single text field?

Reply:
> Maybe you have a mock of the new UI if I misunderstood what do you mean?

Something like this.

![channel serialization mock](https://i.imgur.com/fQGzHCm.png)

Here's a possible grammar in EBNF.
```ebnf
channel = name , space ,
          [ "key:" , key , space ] ,
          buffer size , space ,
          auto clear buffer , space ,
          save , space ,
          enabled , space ,
          detached , space ,
          default modes , newline ;
name = ? irc channel name ? ;
key = ? irc channel key ? ;
buffer size = "default" | integer ;
auto clear buffer = "default" | boolean ;
save = boolean ;
enabled = boolean ;
detached = boolean ;
default modes = ? irc channel modes, may contain spaces ? ;

space = " " , { " " } ;
newline = "\n" ;
boolean = "false" | "true" ;
integer = "0" | "1"..."9" , { "0"..."9" } ;
```

> This will break modules which add parts of UI to that page, e.g. stickychan.

I overlooked that section of the page, my bad. I'll abandon this idea.